lms/djangoapps/teams/csv.py

```diff
@@ -306,6 +310,20 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                 return False
         return True
 
+    def validate_compatible_enrollment_modes(self, user, team_name, teamset_id):
+        """
+        Validates that only students enrolled in a masters track are on a single team. Disallows mixing of masters
+        with other enrollment modes on a single team.
+        """
+        if(teamset_id, team_name) not in self.user_enrollment_by_team:
+            self.user_enrollment_by_team[teamset_id, team_name] = set()
+        self.user_enrollment_by_team[teamset_id, team_name].add(self.user_to_actual_enrollment_mode[user.id])
+        if 'masters' in self.user_enrollment_by_team[teamset_id, team_name] and \
+                len(self.user_enrollment_by_team[teamset_id, team_name]) > 1:
```

Comment:
This length check is fast but semantically meaningless. It would probably read better to have separate lists `protected=('masters') \ unprotected=('audit', 'verified', '*')` and make sure they don't overlap. It carries more meaning and is extensible for new enrollment modes like `bachelors` which might extend the list of protected enrollment modes that could overlap.

Reply:
I like this logic because it's straightforward and lightweight. The user_enrollment_by_team is a set. In python, sets can only contain unique elements, so if the set has 'masters' and is greater than 1 this means students with other enrollment mode are in the course, which is not allowed.

Reply:
I think trying to guess at potential future changes to this behavior is a slippery slope we should avoid

Reply:
Never mind, I see what @nsprenkle was getting at, I believe he was more or less suggesting we try to mimic the `user_organization_protection_status` check in the api.

I think that's a good suggestion. Maybe make a function `is_enrollment_mode_protected` or something like that that checks if a course mode is in `api.ORGANIZATION_PROTECTED_MODES` and then check all(is_enrollment_mode_protected) or not any(is_enrollment_mode_protected)